pkg/search/postgres/store_cache.go

```diff
@@ -199,7 +168,7 @@ func (c *cachedStore[T, PT]) DeleteMany(ctx context.Context, identifiers []strin
 		}
 		filteredIDs = append(filteredIDs, c.pkGetter(obj))
```

Comment:
Should you revive this work, please pay attention to the fact that SAC filtering in the old single-item delete method has an exception for globally-scoped permissions that should be propagated here.